mapflow/mapflow.py

```diff
@@ -2223,6 +2231,13 @@ def preview(self) -> None:
             self.preview_catalog(image_id=image_id)
         else:  # XYZ providers
             self.preview_xyz(provider=provider, image_id=image_id)
+    
+    def preview_or_search(self) -> None:
```

Comment:
"Magnifier" button now functions both ways: for switching to imagery search (if it is selected as datasource) or just as preview (for every other provider).